### PR TITLE
Disable applications page when windows instance isn't ready.

### PR DESCRIPTION
**Pull request by Gentux**
Recent work from @MeoBlodnasir made IaaS API more reliable.
We can use this services to allow/forbid users to interact with features which need windows to be up and running.

### CODE REVIEW

File: webapp/website/ts/components/applications/controllers/ApplicationsCtrl.ts [hunk 1]
```diff
@@ -22,24 +22,32 @@
...
 /// <amd-dependency path="./ApplicationCtrl" />
 import { ApplicationsSvc, IApplication } from "../services/ApplicationsSvc";
+import { ServicesFct } from "../../services/services/ServicesFct";
 
 "use strict";
 
 export class ApplicationsCtrl {
 
 	applications: IApplication[];
+	windowsState: boolean;
```
Comment (dynamiccast) [inline webapp/website/ts/components/applications/controllers/ApplicationsCtrl.ts]: Should be initialized to a default value

File: webapp/website/ts/components/applications/views/applications.html [hunk 2]
```diff
@@ -39,15 +37,13 @@ <h2 class="md-title">Upload to VM</h2>
...
-			<p>
-				<strong>Windows</strong> can take some time to start all services, if you can't access <em>Desktop</em> right now, wait a minute before retrying.
-			</p>
-		</md-card-content>
-	</md-card>
-</div>
+<md-card ng-if="!applicationsCtrl.windowsState">
+	<md-card-content>
+		<p>
+			<strong>Windows</strong> seems to be unavailable, please ensure services is up and running in <strong>Services</strong> page.
```
Comment (dynamiccast) [inline webapp/website/ts/components/applications/views/applications.html]: services **are** up and running

File: webapp/website/ts/components/services/services/ServicesFct.ts [hunk 3]
```diff
@@ -64,6 +64,17 @@ export class ServicesFct {
 		}
 	}
 
+	getWindowsStatus(): angular.IPromise<boolean> {
+		return this.servicesSvc.getAll().then((services: IService[]) => {
+			let state = false;
+			services.forEach(function(service: IService) {
+				if (service.Ico === "windows" && service.Status === "running") {
```
Comment (dynamiccast) [inline webapp/website/ts/components/services/services/ServicesFct.ts:71]: This loop implies all windows must be running is order for state to be on. Should not only one be enough ?

Comment (Gentux) [inline webapp/website/ts/components/services/services/ServicesFct.ts:71]: This loop return true if there is one or more windows services up. It does what it's supposed to do.

Comment (dynamiccast) [inline webapp/website/ts/components/services/services/ServicesFct.ts:71]: My bad

**Comment by dynamiccast** [comment]
Awesome. We will also need to prevent the admin from creating a user if Windows is not running. @Gentux any chance you take care of this ?

**Pull request merged**